packages/common/src/create-icon-set.tsx

```diff
@@ -1,35 +1,79 @@
-import React, { forwardRef, type Ref } from 'react';
+import React, { forwardRef, type Ref, useEffect } from 'react';
 
 import { PixelRatio, Platform, Text, type TextProps, type TextStyle, processColor } from 'react-native';
 
 import NativeIconAPI from './NativeVectorIcons';
 import createIconSourceCache from './create-icon-source-cache';
+import { isDynamicLoadingEnabled } from './dynamicLoading/dynamic-loading-setting';
+import { dynamicLoader } from './dynamicLoading/dynamic-font-loading';
+import type { FontSource } from './dynamicLoading/types';
 import ensureNativeModuleAvailable from './ensure-native-module-available';
 
 export const DEFAULT_ICON_SIZE = 12;
 export const DEFAULT_ICON_COLOR = 'black';
 
+type ValueData = { uri: string; scale: number };
+type GetImageSourceSyncIconFunc<GM> = (name: GM, size?: number, color?: TextStyle['color']) => ValueData | undefined;
+type GetImageSourceIconFunc<GM> = (
+  name: GM,
+  size?: number,
+  color?: TextStyle['color'],
+) => Promise<ValueData | undefined>;
+
 export type IconProps<T> = TextProps & {
   name: T;
   size?: number;
   color?: TextStyle['color'];
   innerRef?: Ref<Text>;
 };
 
-export const createIconSet = <GM extends Record<string, number>>(
+type IconComponent<GM extends Record<string, number>> = React.FC<
+  TextProps & {
+    name: keyof GM;
+    size?: number;
+    color?: TextStyle['color'];
+    innerRef?: Ref<Text>;
+  } & React.RefAttributes<Text>
+> & {
+  getImageSource: GetImageSourceIconFunc<keyof GM>;
+  getImageSourceSync: GetImageSourceSyncIconFunc<keyof GM>;
+};
+
+export type CreateIconSetOptions = {
+  postScriptName: string;
+  fontFilename: string;
+  fontSource?: FontSource;
+  fontStyle?: TextProps['style'];
+};
+
+export function createIconSet<GM extends Record<string, number>>(
   glyphMap: GM,
-  fontFamily: string,
-  fontFile: string,
+  postScriptName: string,
+  fontFilename: string,
   fontStyle?: TextProps['style'],
-) => {
-  // Android doesn't care about actual fontFamily name, it will only look in fonts folder.
-  const fontBasename = fontFile ? fontFile.replace(/\.(otf|ttf)$/, '') : fontFamily;
+): IconComponent<GM>;
+export function createIconSet<GM extends Record<string, number>>(
```

Comment:
I like the change you've made with the object arguments here.
Given we are making a major breaking change with @RNVI anyway and this is typically only used internally by the font libraries, I'm tempted to switch across entirely.

Reply:
This was previously [part of the public api](https://github.com/oblador/react-native-vector-icons?tab=readme-ov-file#createiconsetglyphmap-fontfamily-fontfile) and I believe that keeping it working will make life easier for people who migrate to the new version. People might be depending on this in their apps where it's easy to update but also in their dependencies where it's not. 
Keeping the old signature working could reduce the number of issues people will open about this (=> easier life for maintainers as well). It's just a few lines of code here to support this.

It's true that internally, we would switch to this new signature. I have the diff for that almost ready.